ntp-proto/src/packet.rs

```diff
@@ -113,25 +113,134 @@ pub struct NtpPacket<'a> {
     mac: Option<Mac<'a>>,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Clone, PartialEq, Eq)]
 pub enum ExtensionField<'a> {
-    Unknown { typeid: u16, data: Cow<'a, [u8]> },
+    UniqueIdentifier(Cow<'a, [u8]>),
+    NtsCookie(Cow<'a, [u8]>),
+    NtsCookiePlaceholder {
+        body_length: u16,
+    },
```

Comment:
maybe we should just store the input here? we'll have to see when we start using this field